src/program.rs

```diff
@@ -18,9 +17,9 @@ impl Program {
     /// Create a program from source text.
     // TODO: Make this function return a Result.
```

Comment:
I think now would be a good time to fix this. Just need to add an error state like `Error::InvalidProgram`, and return a result here.